SEED-platform/seed
# add backend and  frontend data aggregation

#### Any background context you want to provide?
As SEED prepares for BPS integration, SEED will be adding functionality to help users track their buildings over time to meet emission goals. To accomplish this building tracking 3 new related models will be added to the DB, 
- Data View 
- Data Aggregation
- Filter Group 

This PR focuses on adding the Data Aggregation model and features. 

Data Aggregations take a selected group of properties and apply a calculation to a specific column. The aggregations are limited to Avg, Count, Max, Min, and Sum. The selected column can be a standard column like 'Total GHG Emissions', extra data columns specified by the user, or derived columns.

Because this is being developed in tandem with Data Views and Filter Groups, this is not the intended final functionality. Instead, it introduces the model and functions to be later incorporated (and modified) with the other proposed models.


#### What's this PR do?
- Adds a new model to the DB, DataAggregation
- Adds back and front end CRUD functionality 
- Adds a data aggregation modal
- Generates the `DataAggregation.evalute()` endpoint
- Adds basic testing

#### How should this be manually tested?
This can be tested with any building import file with more than three entries with integer/float values in a default column and an extra data column.

- Create a derived column
- Go to Inventory > summary > Data Aggregations
- Test CRUD functionality from the Data Aggregation modal 
- Ensure the evaluation values are correct


#### What are the relevant tickets?
#3326 #3329

#### Screenshots (if appropriate)
![Screen Shot 2022-07-08 at 9 28 18 AM](https://user-images.githubusercontent.com/58446472/178023286-4c467788-0353-472f-9ede-3eb786f23052.png)
![Screen Shot 2022-07-08 at 9 28 00 AM](https://user-images.githubusercontent.com/58446472/178023288-1ffeedd8-661e-49e4-acc4-e8dfe339814b.png)
